### ISSUE TITLE
Turn on CPS for VS2017

### ISSUE THREAD

**Issue by aienabled** [Microsoft/VSProjectSystem#173]
Hello!
The guide here https://github.com/Microsoft/VSProjectSystem/blob/master/doc/scenario/turn_on_cps_for_csharp_vb_project.md says we need to switch project type GUID to the CPS one to use it. But I've noticed that if `csproj` defined this way:
```XML
<Project>
...</Project>
```
It will load CPS as well even if project type GUID is kept as `FAE04EC0-301F-11d3-BF4B-00C04F79EFBC`. Also I've noticed that projects created from Class Library (.NET Standard) template are using CPS.

Could you clarify please how VS2017 determines when to use CPS and when to use legacy project system?

Regards!

**Comment by adrianvmsft** [comment]
Indeed, behavior may be different in VS 2017 as now .Net Core is implemented using CPS.
In VS 2017 there is a new mechanism (via the C# project factory) to selectively switch particular C# flavors to CPS while preserving the project type GUIDs defined in the project unchanged (for backward compatibility).
You can find the implementation [here](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/VS?path=/src/vsproject/vsproject/ProjectSelector.cpp&version=GBrel/d15rel&_a=contents&line=21&lineStyle=plain&lineEnd=22&lineStartColumn=1&lineEndColumn=1). 
That may have triggered the difference in behavior you are seeing.

**Comment by wjk** [comment]
@adrianvmsft Hate to say it, but that's a MS-internal-only link you provided. No one outside of Microsoft can see it. Did you mean to link something else?
